packages/kit/src/utils/filesystem.js

```diff
@@ -17,27 +18,60 @@ export function rimraf(path) {
 }
 
 /**
- * @param {string} from
- * @param {string} to
- * @param {(basename: string) => boolean} filter
+ * @param {string} source
+ * @param {string} target
+ * @param {{
+ *   filter?: (basename: string) => boolean;
+ *   replace?: Record<string, string>;
+ * }} opts
  */
-export function copy(from, to, filter = () => true) {
-	if (!fs.existsSync(from)) return [];
-	if (!filter(path.basename(from))) return [];
+export function copy(source, target, opts = {}) {
+	if (!fs.existsSync(source)) return [];
 
+	/** @type {string[]} */
 	const files = [];
-	const stats = fs.statSync(from);
-
-	if (stats.isDirectory()) {
-		fs.readdirSync(from).forEach((file) => {
-			files.push(...copy(path.join(from, file), path.join(to, file)));
-		});
-	} else {
-		mkdirp(path.dirname(to));
-		fs.copyFileSync(from, to);
-		files.push(to);
+
+	const prefix = posixify(target) + '/';
+
+	const regex = opts.replace
+		? new RegExp(`\\b(${Object.keys(opts.replace).join('|')})\\b`, 'g')
+		: null;
+
+	/**
+	 * @param {string} from
+	 * @param {string} to
+	 */
+	function go(from, to) {
```

Comment:
`go` sounds like `cd`. would something like `do_copy` be accurate?

Reply:
i have an irrational aversion to the `do_foo` naming convention. in my mind 'go' is like 'we've finished setup, now we can go'